### ISSUE TITLE
Gym Entry Exams

### ISSUE THREAD

**Issue by Archenemis** [ObliqueNET/Stratos#277]
So. Had an idea, to cut out all the fluff and get to the point: I wanted to have some words posted somewhere to help give people some helpful guidance and help to decrease Toxicity. Lone desired for a system for players to queue into. And Lex brought up the idea of making players take a quiz to get into the gyms. (Joke or not, it has merit).

So the combined idea is to make it so that players can talk to an NPC (That or receive some sort of text similarly to when you buy claims at /warp buyclaims) When they do so. They have to pass a short quiz. Passing the short quiz and showing that they can look up mons and are capable of doing some basic research on their own team enables them to enter into a sort of queue.  The queue pings relevant gym leaders that are online for the gym that they are waiting for.

While in this queue. It would be nice if there was a way to show to the player that is waiting a list of relevant rules for the gym as well as some helpful words. Something along the lines of "We've received your challenge. An Available Gym Leader will be with you when next available. If you find yourself having trouble fighting the Gyms, don't be afraid to ask the gym leader's any questions you may have. They are here to fight you with everything they have and to present a challenge to you. Not to block your progress. They can help you to improve as a player."

I believe Lone's intent with the Queue system as well is to enable the players to see how long they have left on their cooldown to challenge as well. The system Lone is desiring to be in place will keep there from being backup and cut down on arguing of who was there first (May also incidentally prevent people from viewing fights before they try them since they'll be free to go and do their own thing and be notified when it's their turn) and enable gym leaders who may not be online, but still available to be notified of a challenger without having a staff member or player message them over discord.

Another thing. Whether it's through this system or not. Or even just a place holder for now. I think it would be important to give the message to newer players to basically say "The Gym Leaders are your rivals, but they're not your enemies. They want to see you grow. If you are having trouble with beating a gym, or are getting frustrated. Consider asking the Gym Leaders themselves on advice. They will gladly help"

Perhaps it's fairly easy to see the intent, but it's to bring about less toxicity and to promote more sportsmanlike behavior. To try and help players to view the rewards for beating the gyms as a secondary rewards to actually beating them. It won't be an immediately noticeable change, and in fact it's something that we won't really even be able to know for sure if it's working. There will still be cheaters and this won't instantly change all of their minds. If it even changes 1/10 challengers minds about grabbing a pokemon someone is marketing to them as a "Guaranteed to beat that gym pokemon" or stops them from going to discord and trying to share teams that were used to beat a gym.. Then it will have been worth it.

Lone is coming up with a few commands that could possibly be run alongside of this, but this system could end up making the revamped Stratos feel even more sleek and help it stand out even more.

Also, here is a few example questions we could using for the NPC that screens players before they enter the queue.

Quiz questions

Which of these moves is allowed in a gym fight?
A) Baton Pass
B) Minimize
C) Horn Drill
D) Thief

Which of the following Pokemon is Uber?
A) Blaziken (Blaze)
B) Charizard (Mega X)
C) Garchomp (Rough Skin)
D) Ash-Greninja

What's the level limit of the first Gym?
A)25
B)35
C)45
D)30

At what point can you use Mega Pokemon?
A) Immediately
B) During your turn
C) After the 2nd Gym
D) After the 5th gym.

Which of these pokemon is legal to use in the gyms?
A) Buzzwole
B) Articuno
C) Kommo-o
D) Aegislash

All in all. These things combined should be capable of Cutting down on the pointless battles, quick disqualifications from the gym, toxicity, bans, and wait times while at the same time promoting organization, sportsmanship, positive attitudes, and I feel it will help give a sense of professionalism with how the gyms are done as well.

The win and loss commands could be tied into this thing if it's desired, but I think it's fine to just keep it seperate, just enable the queue for a gym to only be allowed the previous is beaten.

**Comment by CenizaRey** [comment]
This is a nifty idea but I really feel like you could’ve condensed this thread more. But yeah, I suggested this a while back and everyone agreed it was a cool idea but nothing came of it /shrug

**Comment by Archenemis** [comment]
Mainly wanted to convey it isn’t my idea alone, and try and explain the reasoning behind each part. I do tend to over explain.

A lot of this would also need to be ran by hiro to have it working on full effect but also..

TLDR: we should consider using a quiz, queue system, and giving players some positive inspiration when approaching the gyms. It would promote healthy and positive attitudes for both staff and challengers.

**Comment by Rasgnarok** [comment]
I wholeheartedly agree with this part: 

> Whether it's through this system or not. Or even just a place holder for now. I think it would be important to give the message to newer players to basically say "The Gym Leaders are your rivals, but they're not your enemies. They want to see you grow. If you are having trouble with beating a gym, or are getting frustrated. Consider asking the Gym Leaders themselves on advice. They will gladly help"

Despite it being the intent, there should be a better way of expressing it. Open to suggestions on how to convey that, beyond the speech to existing gym leaders.

**Comment by Rasgnarok** [comment]
> But yeah, I suggested this a while back and everyone agreed it was a cool idea but nothing came of it /shrug

I don't remember agreeing to this, which probably can explain why it is not in action right now. 

I have a couple issues with a quiz, namely that it feels like a school test where the server is meant to be an escape from duties and homework, and that a lot of these can be also googled for an answer, or shouted in chat until someone replies to them. Now you might say 'let's then enforce a rule they can't ask about the quiz', and I would go back to this: how well did that strategy work for Terra's quest questions?

**Comment by Archenemis** [comment]
First to address the part you're interested in: I initially was thinking on having it as floating text inside of at least the electric gym or perhaps listed somewhere in /gyms. But to be honest. Most problematic people don't even read /gyms in the first place. But if you're fighting someone and the text for it is on your screen the entire time you fight them. You're more prone to give it more than a passing glance during the fight.

----------------------------------------------------
I would like to mention ways to deal with the quiz though. If you really don't think it's a good idea, then feel free to ignore this.

Truth be told, enforcing a rule about the quiz itself would be self defeating. It creates the same problem that the gyms already have just in a different area. So I don't think that enforcing it is the right answer, instead let's make it a "False Quiz" Enabling them to just talk through the quiz and select any of the answers to unlock the feature, (Or take on the first gym) but then be shown which they got wrong and what the correct answer would have been.

Could also go a step further and actually make the quiz "Open Book" as well as randomized and on a short timer. Making it about.. 10-20 seconds to answer each question. No time to ask in chat. No way to have your friends feed you the answers if you don't know them. You'll just have to click through it and guess what you know. But at the end be surprised to see you passed anyways.. I also think it would be a nice touch to have the tester give a comment based on how many you got right: "I mean, congrats. If you really want to call that a win." "You could do better." "You really wanna settle for this score?" "You're gonna go far kid."

The point of the quiz would then be more so to show them what they don't know, rather than to make them guess right and research. If they didn't know some of these basic questions. They should consider reading up on /gyms and /rules and looking into Smogon OU rules. I also wouldn't consider breaking out google to answer these questions against the rules, as encouraging them to do a little research also helps a lot.

If it is in this style though, could also consider shortening it down to 3 questions.

Players likely wouldn't bother with giving them answers if they did manage to get them out, What's the point? You win either way, and by the time they get the answer, it's likely no longer relevant.

Have to admit though, a lot of this is psychological, but it can be effective.

Incidentally randomizing the quiz and adding a timer to it could be a bit tedious, but it could also be useful for the creation of other events and re purposed for more things later down the road.

If you do get it fully randomized and setup with a timer on it. Might even consider giving a slight reward for getting a perfect score. If the Tester is like "Well, I had thought to give you a little something if you did well enough." Players that notice that wouldn't be satisfied with a sub-par score.

One last thing to mention though: It is an escape from real life duties. However Pokemon is in of itself a game of tactics, calculations, decision making, and mind gaming. Pokemon is a mental game of sorts already. People would be fine with an optional quiz. Those that seek entry into the gyms would understand more quickly the rule sets and that the gyms are competitive and approach with a group of 6 at level 35 to the very least.

TLDR: I think the text should go somewhere in the gyms or in the rules. Randomized quiz with a short timer would solve cheating. Let players pass regardless of score, but offer a little something for getting a perfect score. Incidentally I didn't mean to over explain again either.

**Comment by TheDovahkiin74** [comment]
I like this a lot, it would definitely cut down on the dq's that are constantly being given out, while making sure that the players know what they are doing, another thing I'd like to add I'd at the end of the test tell them about smogon. I used thst site a lot when I first started out in competitive battling. Showdown may also be another suggestion to give, as I still use it to test things like stats and how well things go together.
